bfdd/bfd_packet.c

```diff
@@ -650,8 +651,16 @@ int bfd_recv_cb(struct thread *t)
 				 bfd->mh_ttl, ttl);
 			return 0;
 		}
-	} else if (bfd->local_address.sa_sin.sin_family == AF_UNSPEC) {
-		bfd->local_address = local;
+	/* Check if key.local is all 0's (unassigned) */
+	} else if (memcmp(&bfd->key.local, &zero_addr,
```

Comment:
I am not sure about this. We didn't modify `key.local` previously so why should we start doing this now? This change can definitely affect something else and the whole code should be checked very carefully before doing this.